pages/bluedialog/bluedialog.page.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect, useRef } from 'react';
+
+import {
+  Box,
+  Button,
+  DatePicker,
+  ExpandableSection,
+  Form,
+  FormField,
+  Header,
+  Input,
+  Select,
+  SpaceBetween,
+  Textarea,
+} from '~components';
+
+import styles from './styles.scss';
+
+export default function Bluedialogbox() {
+  const dialogRef = useRef<HTMLDivElement>(null);
+  const triggerRef = useRef<HTMLElement | null>(null);
+  // Inputs (these following can be clubbed into one state object)
+  const [dateValue, setDateValue] = React.useState('');
+  const [amountValue, setAmountValue] = React.useState('0');
+  const [descriptionValue, setDescriptionValue] = React.useState('');
+  const [selectedService, setSelectedService] = React.useState({ label: 'Cloudwatch', value: 'Cloudwatch' });
+
+  useEffect(() => {
+    // Store the element that had focus before dialog opened
+    triggerRef.current = document.activeElement as HTMLElement;
```

Comment:
If the dialog is opened programmatically this transition can be weird. Would it be better to handle the focus transition on the outside? If the parent component provides onSubmit/onCancel then the focus transition can be done as part of it:

```
function ParentComponent() {
  // ...
  
  function returnFocus() {
    sendButtonRef.current?.focus()
  }
  
  function onSubmit() {
    // submit
    returnFocus();
  }
  
  function onCancel() {
    // cancel
    returnFocus();
  }
  
  return (
    <div>
      // ...
      <Bluedialog onSubmit={onSubmit} onCancel={onCancel} />
    </div>
  );
}
```

Reply:
Like I mentioned previously, this is just a code snippet. It is upto the individual users on how they want to use it.